### PR TITLE
ceph-crash: use client.crash[.host] to post, and provsion keys via mgr/ssh + ceph-daemon

### PR DESCRIPTION
**Pull request by liewegas**
Currently ceph-crash assumes that a client.admin key is present on the host to post crash dumps.  That's usually not the case.

We'll need to backport the mon/MonCap and ceph-crash patches to nautilus, and update ceph-ansible to deploy a client.crash keyring at /etc/ceph/ceph.client.crash.keyring, using something like::

ceph get-or-create-key client.crash mon 'allow profile crash' mgr 'allow profile crash'

### CODE REVIEW

**Comment by liewegas** [comment]
> Shouldn't we have a bootstrap crash key generated by the mons by default?

For ceph-daemon/ssh orch, my plan was to have a per-host key, so that an individual host could be removed without changing the keys globally.  I'd like to get away from the single-bootstrap-key-per-cluster model in general.  The ceph-daemon+ssh, for instance, could use a single-use key for deploying OSDs, and doesn't need a bootstrap key at all for creating the other daemons since the mgr is driving the process remotely.

Long story short, I think ceph-ansible can make it's own choice here.. either create a single global key for simplicity, or generate a per-host one.  ceph-crash will try client.crash.$host, then client.crash, then client.admin, in that order.

**Comment by leseb** [comment]
> > Shouldn't we have a bootstrap crash key generated by the mons by default?
> 
> For ceph-daemon/ssh orch, my plan was to have a per-host key, so that an individual host could be removed without changing the keys globally. I'd like to get away from the single-bootstrap-key-per-cluster model in general. The ceph-daemon+ssh, for instance, could use a single-use key for deploying OSDs, and doesn't need a bootstrap key at all for creating the other daemons since the mgr is driving the process remotely.
> 
> Long story short, I think ceph-ansible can make it's own choice here.. either create a single global key for simplicity, or generate a per-host one. ceph-crash will try client.crash.$host, then client.crash, then client.admin, in that order.

I'm fine with that, I'll improve https://github.com/rook/rook/pull/3316 once this merges. As for ceph-ansible, I don't know so highlighting @guits FYI.
Thanks!

**Comment by tchaikov** [comment]
see https://tracker.ceph.com/issues/42369
